diff -up chromium-94.0.4606.81/build/config/compiler/BUILD.gn.i686-hack chromium-94.0.4606.81/build/config/compiler/BUILD.gn
--- chromium-94.0.4606.81/build/config/compiler/BUILD.gn.i686-hack	2021-10-20 15:56:09.409349920 -0400
+++ chromium-94.0.4606.81/build/config/compiler/BUILD.gn	2021-10-20 15:57:30.409831217 -0400
@@ -360,6 +360,9 @@ config("compiler") {
     if (fatal_linker_warnings && is_apple) {
       ldflags += [ "-Wl,-fatal_warnings" ]
     }
+
+    ldflags += [ "-Wl,--no-keep-memory" ]
+    ldflags += [ "-Wl,--reduce-memory-overheads" ]
   }
 
   if (is_clang && is_debug) {
