diff -up chromium-114.0.5735.26/sandbox/policy/linux/bpf_network_policy_linux.cc.me chromium-114.0.5735.26/sandbox/policy/linux/bpf_network_policy_linux.cc
--- chromium-114.0.5735.26/sandbox/policy/linux/bpf_network_policy_linux.cc.me	2023-05-13 12:09:44.423727385 +0200
+++ chromium-114.0.5735.26/sandbox/policy/linux/bpf_network_policy_linux.cc	2023-05-13 17:52:19.934347246 +0200
@@ -11,7 +11,6 @@
 #include <linux/net.h>
 #include <linux/netlink.h>
 #include <linux/sockios.h>
-#include <linux/wireless.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <sys/inotify.h>
@@ -48,6 +47,10 @@ using sandbox::syscall_broker::BrokerPro
 #define F2FS_IOC_GET_FEATURES _IOR(0xf5, 12, uint32_t)
 #endif
 
+#if !defined(SIOCGIWNAME)
+#define SIOCGIWNAME  0x8B01
+#endif
+
 namespace sandbox::policy {
 
 namespace {
