diff -up chromium-86.0.4240.111/base/strings/string_util_internal.h.ranges chromium-86.0.4240.111/base/strings/string_util_internal.h
--- chromium-86.0.4240.111/base/strings/string_util_internal.h.ranges	2020-10-21 18:25:31.939001816 -0400
+++ chromium-86.0.4240.111/base/strings/string_util_internal.h	2020-10-21 18:26:36.686485073 -0400
@@ -9,6 +9,7 @@
 #include "base/notreached.h"
 #include "base/strings/string_piece.h"
 #include "base/third_party/icu/icu_utf.h"
+#include "base/util/ranges/algorithm.h"
 
 namespace base {
 
@@ -578,8 +579,7 @@ StringType DoReplaceStringPlaceholders(
             ReplacementOffset r_offset(index,
                                        static_cast<int>(formatted.size()));
             r_offsets.insert(
-                std::upper_bound(r_offsets.begin(), r_offsets.end(), r_offset,
-                                 &CompareParameter),
+                util::ranges::upper_bound(r_offsets, r_offset, &CompareParameter),
                 r_offset);
           }
           if (index < substitutions)
