diff -up chromium-99.0.4844.51/base/allocator/partition_allocator/starscan/metadata_allocator.h.el7-extra-operator-equalequal chromium-99.0.4844.51/base/allocator/partition_allocator/starscan/metadata_allocator.h
--- chromium-99.0.4844.51/base/allocator/partition_allocator/starscan/metadata_allocator.h.el7-extra-operator-equalequal	2022-03-05 18:37:25.979241754 -0500
+++ chromium-99.0.4844.51/base/allocator/partition_allocator/starscan/metadata_allocator.h	2022-03-05 18:37:25.979241754 -0500
@@ -39,6 +39,11 @@ class MetadataAllocator {
   }
 
   template <typename U>
+  bool operator==(const MetadataAllocator<U>&) const {
+    return true;
+  }
+
+  template <typename U>
   bool operator!=(const MetadataAllocator<U>& o) {
     return !operator==(o);
   }
diff -up chromium-99.0.4844.51/third_party/pdfium/core/fxcrt/fx_memory_wrappers.h.el7-extra-operator-equalequal chromium-99.0.4844.51/third_party/pdfium/core/fxcrt/fx_memory_wrappers.h
--- chromium-99.0.4844.51/third_party/pdfium/core/fxcrt/fx_memory_wrappers.h.el7-extra-operator-equalequal	2022-03-06 19:14:20.892513442 -0500
+++ chromium-99.0.4844.51/third_party/pdfium/core/fxcrt/fx_memory_wrappers.h	2022-03-06 19:15:40.412060574 -0500
@@ -70,6 +70,7 @@ struct FxPartitionAllocAllocator {
   }
 
   // There's no state, so they are all the same,
+  bool operator==(const FxPartitionAllocAllocator&) const { return true; }
   bool operator==(const FxPartitionAllocAllocator& that) { return true; }
   bool operator!=(const FxPartitionAllocAllocator& that) { return false; }
 };
