diff -up chromium-105.0.5195.125/v8/BUILD.gn.arm64-gcc-fix chromium-105.0.5195.125/v8/BUILD.gn
--- chromium-105.0.5195.125/v8/BUILD.gn.arm64-gcc-fix	2022-09-20 18:12:30.766293126 +0000
+++ chromium-105.0.5195.125/v8/BUILD.gn	2022-09-20 18:13:04.376435347 +0000
@@ -1095,7 +1095,7 @@ config("toolchain") {
     defines += [ "V8_TARGET_ARCH_ARM64" ]
     if (current_cpu == "arm64") {
       # This will enable PAC+BTI in code generation and static code.
-      if (v8_control_flow_integrity &&
+      if (v8_control_flow_integrity && is_clang &&
           (!build_with_chromium || arm_control_flow_integrity == "standard")) {
         cflags += [ "-mbranch-protection=standard" ]
         asmflags = [ "-mmark-bti-property" ]
