diff -up chromium-114.0.5735.26/components/omnibox/browser/omnibox_edit_model.cc.me chromium-114.0.5735.26/components/omnibox/browser/omnibox_edit_model.cc
--- chromium-114.0.5735.26/components/omnibox/browser/omnibox_edit_model.cc.me	2023-05-14 09:14:10.886314480 +0200
+++ chromium-114.0.5735.26/components/omnibox/browser/omnibox_edit_model.cc	2023-05-14 09:16:59.380054720 +0200
@@ -79,7 +79,7 @@
 #include "ui/gfx/vector_icon_types.h"
 #endif
 
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
+#ifdef GOOGLE_CHROME_BRANDING
 #include "components/vector_icons/vector_icons.h"  // nogncheck
 #endif
 
@@ -628,7 +628,7 @@ bool OmniboxEditModel::ShouldShowCurrent
 }
 
 ui::ImageModel OmniboxEditModel::GetSuperGIcon(int image_size, bool dark_mode) {
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
+#ifdef GOOGLE_CHROME_BRANDING
   if (dark_mode) {
     return ui::ImageModel::FromVectorIcon(
         vector_icons::kGoogleGLogoMonochromeIcon, ui::kColorRefPrimary100,
