diff -up chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc.el7-memfd-include chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc
--- chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc.el7-memfd-include	2022-06-15 10:52:49.553817031 -0400
+++ chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc	2022-06-15 10:56:15.775173013 -0400
@@ -56,6 +56,7 @@
 
 #if V8_OS_LINUX
 #include <sys/prctl.h>  // for prctl
+#include <linux/memfd.h> // for MFD_CLOEXEC
 #endif
 
 #if defined(V8_OS_FUCHSIA)
