diff -up chromium-87.0.4280.88/net/dns/dns_server_iterator.h.missing-cstddef chromium-87.0.4280.88/net/dns/dns_server_iterator.h
--- chromium-87.0.4280.88/net/dns/dns_server_iterator.h.missing-cstddef	2020-12-15 15:35:38.935281593 -0500
+++ chromium-87.0.4280.88/net/dns/dns_server_iterator.h	2020-12-15 15:35:59.408368520 -0500
@@ -5,6 +5,7 @@
 #ifndef NET_DNS_DNS_SERVER_ITERATOR_H_
 #define NET_DNS_DNS_SERVER_ITERATOR_H_
 
+#include <cstddef>
 #include <vector>
 
 #include "net/base/net_export.h"
