diff -up chromium-108.0.5359.124/third_party/wayland/src/src/wayland-shm.c.me chromium-108.0.5359.124/third_party/wayland/src/src/wayland-shm.c
--- chromium-108.0.5359.124/third_party/wayland/src/src/wayland-shm.c.me	2022-12-24 11:08:03.212333476 +0100
+++ chromium-108.0.5359.124/third_party/wayland/src/src/wayland-shm.c	2022-12-24 11:08:18.316606155 +0100
@@ -44,7 +44,7 @@
 #include <signal.h>
 #include <pthread.h>
 #include <errno.h>
-#include <fcntl.h>
+#include <linux/fcntl.h>
 
 #include "wayland-os.h"
 #include "wayland-util.h"
diff -up chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc.el7-memfd-include chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc
--- chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc.el7-memfd-include	2022-06-15 10:52:49.553817031 -0400
+++ chromium-102.0.5005.115/v8/src/base/platform/platform-posix.cc	2022-06-15 10:56:15.775173013 -0400
@@ -56,6 +56,7 @@
 
 #if V8_OS_LINUX
 #include <sys/prctl.h>  // for prctl
+#include <linux/memfd.h> // for MFD_CLOEXEC
 #endif
 
 #if defined(V8_OS_FUCHSIA)
diff -up iridium-browser-2022.12.108.1/third_party/wayland/src/cursor/os-compatibility.c.me iridium-browser-2022.12.108.1/third_party/wayland/src/cursor/os-compatibility.c
--- iridium-browser-2022.12.108.1/third_party/wayland/src/cursor/os-compatibility.c.me	2022-12-08 21:59:43.502200984 +0100
+++ iridium-browser-2022.12.108.1/third_party/wayland/src/cursor/os-compatibility.c	2022-12-08 22:13:53.375653343 +0100
@@ -29,7 +29,8 @@
 
 #include <sys/types.h>
 #include <unistd.h>
-#include <fcntl.h>
+#include <linux/fcntl.h> // for F_SEAL_SHRINK, F_ADD_SEALS, F_SEAL_SEAL
+#include <linux/memfd.h> // for MFD_CLOEXEC
 #include <errno.h>
 #include <signal.h>
 #include <string.h>
