diff -up chromium-95.0.4638.69/net/base/test_data_stream.cc.missing-cstring chromium-95.0.4638.69/net/base/test_data_stream.cc
--- chromium-95.0.4638.69/net/base/test_data_stream.cc.missing-cstring	2021-11-14 16:04:59.911270740 +0000
+++ chromium-95.0.4638.69/net/base/test_data_stream.cc	2021-11-14 16:04:59.881272143 +0000
@@ -5,6 +5,7 @@
 #include "net/base/test_data_stream.h"
 
 #include <algorithm>
+#include <cstring>
 
 namespace net {
 
diff -up chromium-95.0.4638.69/net/filter/filter_source_stream_test_util.cc.missing-cstring chromium-95.0.4638.69/net/filter/filter_source_stream_test_util.cc
--- chromium-95.0.4638.69/net/filter/filter_source_stream_test_util.cc.missing-cstring	2021-11-15 15:54:35.895963882 +0000
+++ chromium-95.0.4638.69/net/filter/filter_source_stream_test_util.cc	2021-11-15 15:54:50.342288149 +0000
@@ -4,6 +4,8 @@
 
 #include "net/filter/filter_source_stream_test_util.h"
 
+#include <cstring>
+
 #include "base/bit_cast.h"
 #include "base/check_op.h"
 #include "third_party/zlib/zlib.h"
diff -up chromium-95.0.4638.69/third_party/webrtc/audio/utility/channel_mixer.cc.missing-cstring chromium-95.0.4638.69/third_party/webrtc/audio/utility/channel_mixer.cc
--- chromium-95.0.4638.69/third_party/webrtc/audio/utility/channel_mixer.cc.missing-cstring	2021-11-14 16:05:24.613116022 +0000
+++ chromium-95.0.4638.69/third_party/webrtc/audio/utility/channel_mixer.cc	2021-11-14 16:05:24.614115975 +0000
@@ -8,6 +8,8 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
+#include <cstring>
+
 #include "audio/utility/channel_mixer.h"
 
 #include "audio/utility/channel_mixing_matrix.h"
diff -up chromium-95.0.4638.69/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc.missing-cstring chromium-95.0.4638.69/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc
--- chromium-95.0.4638.69/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc.missing-cstring	2021-11-14 16:05:24.095140235 +0000
+++ chromium-95.0.4638.69/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc	2021-11-14 16:05:24.095140235 +0000
@@ -19,6 +19,7 @@
 #include <sys/mman.h>
 #include <sys/syscall.h>
 
+#include <cstring>
 #include <memory>
 #include <utility>
 
diff -up chromium-95.0.4638.69/third_party/webrtc/modules/video_coding/utility/ivf_file_reader.cc.missing-cstring chromium-95.0.4638.69/third_party/webrtc/modules/video_coding/utility/ivf_file_reader.cc
--- chromium-95.0.4638.69/third_party/webrtc/modules/video_coding/utility/ivf_file_reader.cc.missing-cstring	2021-11-14 16:05:23.794154305 +0000
+++ chromium-95.0.4638.69/third_party/webrtc/modules/video_coding/utility/ivf_file_reader.cc	2021-11-14 16:05:23.792154399 +0000
@@ -10,6 +10,7 @@
 
 #include "modules/video_coding/utility/ivf_file_reader.h"
 
+#include <cstring>
 #include <string>
 #include <vector>
 
