diff -up chromium-108.0.5359.124/build/config/compiler/BUILD.gn.me chromium-108.0.5359.124/build/config/compiler/BUILD.gn
--- chromium-108.0.5359.124/build/config/compiler/BUILD.gn.me	2023-01-03 21:02:16.602183035 +0100
+++ chromium-108.0.5359.124/build/config/compiler/BUILD.gn	2023-01-03 21:04:11.606320859 +0100
@@ -589,15 +589,6 @@ config("compiler") {
       if (is_clang) {
         cflags_cc += [ "-fno-trigraphs" ]
       }
-    } else if (is_linux) {
-      # TODO(crbug.com/1284275): Switch to C++20 on all platforms.
-      if (is_clang) {
-        cflags_cc += [ "-std=${standard_prefix}++20" ]
-      } else {
-        # The gcc bots are currently using GCC 9, which is not new enough to
-        # support "c++20"/"gnu++20".
-        cflags_cc += [ "-std=${standard_prefix}++2a" ]
-      }
     } else {
       cflags_cc += [ "-std=${standard_prefix}++17" ]
     }
