diff -up chromium-108.0.5359.124/third_party/wayland/src/src/scanner.c.me chromium-108.0.5359.124/third_party/wayland/src/src/scanner.c
--- chromium-108.0.5359.124/third_party/wayland/src/src/scanner.c.me	2022-12-27 16:14:22.507496575 +0100
+++ chromium-108.0.5359.124/third_party/wayland/src/src/scanner.c	2022-12-27 16:35:17.617573610 +0100
@@ -26,6 +26,7 @@
  */
 
 #include "wayland-version.h"
+#include "config.h" // HAVE_STRNDUP
 
 #include <stdbool.h>
 #include <stdio.h>
