diff -up chromium-100.0.4896.75/media/capabilities/webrtc_video_stats_db.h.missing-vector chromium-100.0.4896.75/media/capabilities/webrtc_video_stats_db.h
--- chromium-100.0.4896.75/media/capabilities/webrtc_video_stats_db.h.missing-vector	2022-04-11 18:10:09.979195562 -0400
+++ chromium-100.0.4896.75/media/capabilities/webrtc_video_stats_db.h	2022-04-11 18:10:30.261338074 -0400
@@ -7,6 +7,7 @@
 
 #include <memory>
 #include <string>
+#include <vector>
 
 #include "base/callback_forward.h"
 #include "base/check.h"
